src/compiler/program.ts

```diff
@@ -772,6 +772,28 @@ namespace ts {
         mtime: Date;
     }
 
+    export function readDefaultTypeDirectiveNames(rootPath: string, sys: System): string[] {

```

Comment:
do not think you need `System` here. just `{ directoryExists(path: string): boolean; getDirectories(path: string): string[];}`